### ISSUE TITLE
Destroy rust pointer make the app crash in XR

### ISSUE THREAD

**Issue by hanwencheng** [paritytech/parity-signer#645]
According to a crash report on iphone XR.

That each time when the pointer destroyed, the app crash.

crashed report as followings

```
Thread 0 Crashed:
0   libsystem_kernel.dylib        	0x00000001b4852df0 __pthread_kill + 8
1   libsystem_pthread.dylib       	0x00000001b4772930 pthread_kill + 228 (pthread.c:1458)
2   libsystem_c.dylib             	0x00000001b47008b4 abort + 104 (abort.c:110)
3   libsystem_malloc.dylib        	0x00000001b4763fdc malloc_vreport + 564 (malloc_printf.c:183)
4   libsystem_malloc.dylib        	0x00000001b47641a4 malloc_report + 64 (malloc_printf.c:192)
5   libsystem_malloc.dylib        	0x00000001b4757d1c free + 436 (malloc.c:1733)
6   NativeSigner                  	0x00000001028a3b78 __rust_dealloc + 8
7   NativeSigner                  	0x0000000102884990 specialized EthkeyBridge.destroyDataRef(_:resolve:reject:) + 60 (EthkeyBridge.swift:226)
8   NativeSigner                  	0x0000000102883e24 @objc EthkeyBridge.randomPhrase(_:resolve:reject:) + 120
9   CoreFoundation                	0x00000001b4a72d14 __invoking___ + 148
10  CoreFoundation                	0x00000001b493bde8 -[NSInvocation invoke] + 400 (NSForwarding.m:3312)
11  CoreFoundation                	0x00000001b493ca10 -[NSInvocation invokeWithTarget:] + 80 (NSForwarding.m:3418)
12  React                         	0x0000000102f9c4e4 -[RCTModuleMethod invokeWithBridge:module:arguments:] + 460 (RCTModuleMethod.mm:569)
13  React                         	0x0000000102f9e5e0 facebook::react::invokeInner(RCTBridge*, RCTModuleData*, unsigned int, folly::dynamic const&) + 244 (RCTNativeModule.mm:108)
14  React                         	0x0000000102f9e344 invocation function for block in facebook::react::RCTNativeModule::invoke(unsigned int, folly::dynamic&&, int) + 88 (RCTNativeModule.mm:73)
15  libdispatch.dylib             	0x00000001b470aec4 _dispatch_call_block_and_release + 32 (init.c:1408)
16  libdispatch.dylib             	0x00000001b470c33c _dispatch_client_callout + 20 (object.m:495)
17  libdispatch.dylib             	0x00000001b4718600 _dispatch_main_queue_callback_4CF + 832 (inline_internal.h:2484)
18  CoreFoundation                	0x00000001b49e741c __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16 (CFRunLoop.c:1749)
19  CoreFoundation                	0x00000001b49e2034 __CFRunLoopRun + 1708 (CFRunLoop.c:3069)
20  CoreFoundation                	0x00000001b49e1660 CFRunLoopRunSpecific + 480 (CFRunLoop.c:3192)
21  GraphicsServices              	0x00000001bedf2604 GSEventRunModal + 164 (GSEvent.c:2246)
22  UIKitCore                     	0x00000001b8bb615c UIApplicationMain + 1944 (UIApplication.m:4820)
23  NativeSigner                  	0x0000000102883088 main + 88 (main.m:22)
24  libdyld.dylib                 	0x00000001b485d1ec start + 4
```

**Comment by kirushik** [comment]
@gww-parity I'm not sure if you're following the issues here or not, but this might be something relevant to your current area of work...

**Comment by hanwencheng** [comment]
should be closed by #660.
